### ISSUE TITLE
Opening certain files causes a NullPointerException

### ISSUE THREAD

**Issue by fhomolka** [kevinferrare/spice86#14]
Version tested: 1.29.0

I have attempted to run 3 games: PoP, Dune and Monkey Island. And all of them spit out different errors, but they all fail at the same line.

Prince:
```
[Executor] s.e.l.ExecutableFileLoader - Program entry point is 0x16D0:0x2C4
[Executor] s.e.i.dos.DosInt21Handler - GET DOS VERSION
[Executor] s.e.i.dos.DosInt21Handler - MODIFY MEMORY BLOCK size=10159, blockSegment=477
[Executor] s.e.i.dos.DosInt21Handler - GET DOS VERSION
[Executor] s.e.i.dos.DosInt21Handler - GET INTERRUPT VECTOR INT 0x0, got 0x0:0x0
[Executor] s.e.i.dos.DosInt21Handler - SET INTERRUPT VECTOR FOR INT 0x0 at address 0x16D0:0x366
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET CURRENT DEFAULT DRIVE
[Executor] s.e.i.dos.DosInt21Handler - GET CURRENT DIRECTORY responseAddress=0x198C:0x73FB
[Executor] s.e.i.dos.DosInt21Handler - OPEN FILE C:\prince.dat with mode 0x0 (rwAccessMode:0x0)
[Executor] s.e.i.dos.DosFileManager - Opening file ./PRINCE.DAT with mode r
[Executor] s.e.i.dos.DosInt21Handler - READ FROM FILE handle 5 length 6 to 0x198C:0x743C
[Executor] s.e.i.dos.DosFileManager - Reading from file C:\prince.dat
[Executor] s.e.i.dos.DosInt21Handler - ALLOCATE MEMORY BLOCK size=39
[Executor] s.e.i.dos.DosInt21Handler - MOVE FILE POINTER USING HANDLE. originOfMove=0, fileHandle=5, offset=25086
[Executor] s.e.i.dos.DosFileManager - Moving in file C:\prince.dat
[Executor] s.e.i.dos.DosInt21Handler - READ FROM FILE handle 5 length 514 to 0x298D:0x5E
[Executor] s.e.i.dos.DosFileManager - Reading from file C:\prince.dat
[Executor] s.e.i.dos.DosInt21Handler - FIND FIRST MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./DIGISND3.DAT
[Executor] s.e.i.dos.DosInt21Handler - GET DTA (DISK TRANSFER ADDRESS) DS:DX 0x1DD:0x80
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./SNDDRVRS/PRESET33.DEF
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./VDUNGEON.DAT
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./MIDISND1.DAT
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./DIGISND1.DAT
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./IBM_SND2.DAT
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./DIGISND2.DAT
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./MIDISND2.DAT
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - Found matching file ./IBM_SND1.DAT
[Executor] s.e.i.dos.DosInt21Handler - FIND NEXT MATCHING FILE attributes=0x8, fileSpec=C:????????.???
[Executor] s.e.i.dos.DosFileManager - No more files matching ????????.??? in path ./
[Executor] s.e.i.dos.DosInt21Handler - DOS operation failed with an error. Int will return to null. State is cycles=1682 CS:IP=0xF000:0x28/0xF000:0x28 AX=0x4F43 BX=0x80 CX=0x8 DX=0x0 SI=0x745F DI=0xC BP=0x7444 SP=0x73EC SS=0x198C DS=0x198B ES=0x1DD FS=0x0 GS=0x0 flags=0x7046 (     Z P )
[Executor] s.e.i.dos.DosInt21Handler - OPEN FILE setup.dat with mode 0x0 (rwAccessMode:0x0)
[Executor] spice86.ui.Spice86Application - An error occurred during execution
java.lang.NullPointerException: null
        at java.base/java.io.File.<init>(File.java:278)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:450)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:456)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:456)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toHostCaseSensitiveFileName(DosFileManager.java:420)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.openFile(DosFileManager.java:102)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.openFile(DosInt21Handler.java:290)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.lambda$fillDispatchTable$3(DosInt21Handler.java:71)
        at spice86.emulator.callback.IndexBasedDispatcher.run(IndexBasedDispatcher.java:21)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.run(DosInt21Handler.java:96)
        at spice86.emulator.callback.IndexBasedDispatcher.run(IndexBasedDispatcher.java:21)
        at spice86.emulator.cpu.Cpu.callback(Cpu.java:1793)
        at spice86.emulator.cpu.Cpu.grp4(Cpu.java:1503)
        at spice86.emulator.cpu.Cpu.execOpcode(Cpu.java:1135)
        at spice86.emulator.cpu.Cpu.executeNextInstruction(Cpu.java:200)
        at spice86.emulator.machine.Machine.runLoop(Machine.java:300)
        at spice86.emulator.machine.Machine.run(Machine.java:290)
        at spice86.emulator.ProgramExecutor.run(ProgramExecutor.java:51)
        at spice86.ui.Spice86Application.lambda$startMachine$3(Spice86Application.java:55)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
 ```
        
Dune:
 ```
[Executor] s.e.l.ExecutableFileLoader - Program entry point is 0x1ED:0x0
[Executor] s.e.i.dos.DosInt21Handler - GET CURRENT DEFAULT DRIVE
[Executor] s.e.i.dos.DosInt21Handler - GET/SET CTRL-C FLAG
[Executor] s.e.i.dos.DosInt21Handler - GET/SET CTRL-C FLAG
[Executor] s.e.i.dos.DosInt21Handler - OPEN FILE dunevga.hsq with mode 0x0 (rwAccessMode:0x0)
[Executor] spice86.ui.Spice86Application - An error occurred during execution
java.lang.NullPointerException: null
        at java.base/java.io.File.<init>(File.java:278)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:450)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:456)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:456)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toHostCaseSensitiveFileName(DosFileManager.java:420)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.openFile(DosFileManager.java:102)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.openFile(DosInt21Handler.java:290)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.lambda$fillDispatchTable$3(DosInt21Handler.java:71)
        at spice86.emulator.callback.IndexBasedDispatcher.run(IndexBasedDispatcher.java:21)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.run(DosInt21Handler.java:96)
        at spice86.emulator.callback.IndexBasedDispatcher.run(IndexBasedDispatcher.java:21)
        at spice86.emulator.cpu.Cpu.callback(Cpu.java:1793)
        at spice86.emulator.cpu.Cpu.grp4(Cpu.java:1503)
        at spice86.emulator.cpu.Cpu.execOpcode(Cpu.java:1135)
        at spice86.emulator.cpu.Cpu.executeNextInstruction(Cpu.java:200)
        at spice86.emulator.machine.Machine.runLoop(Machine.java:300)
        at spice86.emulator.machine.Machine.run(Machine.java:290)
        at spice86.emulator.ProgramExecutor.run(ProgramExecutor.java:51)
        at spice86.ui.Spice86Application.lambda$startMachine$3(Spice86Application.java:55)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

Monkey:
```
[Executor] s.e.l.ExecutableFileLoader - Program entry point is 0x12B1:0xE
[Executor] s.e.i.dos.DosInt21Handler - GET DOS VERSION
[Executor] s.e.i.dos.DosInt21Handler - MODIFY MEMORY BLOCK size=10659, blockSegment=477
[Executor] s.e.i.dos.DosInt21Handler - GET DOS VERSION
[Executor] s.e.i.dos.DosInt21Handler - GET INTERRUPT VECTOR INT 0x0, got 0x0:0x0
[Executor] s.e.i.dos.DosInt21Handler - SET INTERRUPT VECTOR FOR INT 0x0 at address 0x114C:0x57CF
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET DEVICE INFORMATION
[Executor] s.e.i.dos.DosInt21Handler - GET CURRENT DEFAULT DRIVE
[Executor] s.e.i.dos.DosInt21Handler - SET INTERRUPT VECTOR FOR INT 0x24 at address 0x114C:0x7A21
[Executor] s.e.i.dos.DosInt21Handler - OPEN FILE 000.lfl with mode 0x0 (rwAccessMode:0x0)
[Executor] spice86.ui.Spice86Application - An error occurred during execution
java.lang.NullPointerException: null
        at java.base/java.io.File.<init>(File.java:278)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:450)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:456)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toCaseSensitiveFileName(DosFileManager.java:456)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.toHostCaseSensitiveFileName(DosFileManager.java:420)
        at spice86.emulator.interrupthandlers.dos.DosFileManager.openFile(DosFileManager.java:102)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.openFile(DosInt21Handler.java:290)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.lambda$fillDispatchTable$3(DosInt21Handler.java:71)
        at spice86.emulator.callback.IndexBasedDispatcher.run(IndexBasedDispatcher.java:21)
        at spice86.emulator.interrupthandlers.dos.DosInt21Handler.run(DosInt21Handler.java:96)
        at spice86.emulator.callback.IndexBasedDispatcher.run(IndexBasedDispatcher.java:21)
        at spice86.emulator.cpu.Cpu.callback(Cpu.java:1793)
        at spice86.emulator.cpu.Cpu.grp4(Cpu.java:1503)
        at spice86.emulator.cpu.Cpu.execOpcode(Cpu.java:1135)
        at spice86.emulator.cpu.Cpu.executeNextInstruction(Cpu.java:200)
        at spice86.emulator.machine.Machine.runLoop(Machine.java:300)
        at spice86.emulator.machine.Machine.run(Machine.java:290)
        at spice86.emulator.ProgramExecutor.run(ProgramExecutor.java:51)
        at spice86.ui.Spice86Application.lambda$startMachine$3(Spice86Application.java:55)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

I have tried using both Oracle Java and OpenJDK, both version 17.
I will admit, I am not the most knowledgeable on Java, and this could be an issue caused by me.
My process was:

1. Download the package from `packages`
2. Run the jar as described in README.md

OS:
Linux 5.10.79-1-MANJARO

**Comment by kevinferrare** [comment]
Hi @fhomolka, thanks for the input!
This is not normal, especially for dune for which I developed this tool and which I have running.
I will look into that!

**Comment by fhomolka** [comment]
Small update:
I tried running on a different machine, and it works well. (aside from the window being too large, but that's easily resolved)
I ran Dune, since that's what this tool was originally made for.

What info can I provide, that would be helpful in resolving this?

**Comment by kevinferrare** [comment]
OK that's reassuring, I was thinking that I didnt test it enough in linux :)

The bug is in the algorithm that tries to convert DOS case insensitive paths to host path (case sensitive for linux).

It seems it goes up 3 folders in the path and encounters an issue, not sure what it is.
It would be nice if you could provide:
 - The full path to the expected file in your machine if you know it
 - The command line you used to launch the program
 - The path from where you launched the program

Thank you in advance!

**Comment by kevinferrare** [comment]
I pushed some code in the latest version that should give you a file not found error with some details instead of the exception.

**Comment by fhomolka** [comment]
Oh, I'm sorry, I missed your previous reply! And it's a shame it did, because it probably would've helped...

I just tried out the new version, and it worked.
So, I decided to investigate further.

Somewhat relieved and ashamed to say that the cause is user error.

I put spice86 in the same directory as my DOS executable.

The command I previously used was:
`java -jar ./spice86-1.28.0.jar DUNEPRG.EXE`
Which correctly spots the EXE, but fails to load any other files.

However, if the command used is
`java -jar ./spice86-1.28.0.jar ./DUNEPRG.EXE`
It also finds all the other files.

So... Issue resolved?
I'm still confused as to how it finds the executable, but not the other files. I might dig into the java code to better understand that.

Thank you for your patience!
